src/app/controllers/authentication.server.controller.js

```diff
@@ -227,47 +226,20 @@ exports.sendOtp = async function (req, res) {
   // 2. Return success statement to front end
 
   let otp = res.locals.otp
-  let email = res.locals.email
+  let recipient = res.locals.email
 
-  let emailHTML
   try {
-    emailHTML = await renderPromise(res, 'templates/otp-email', {
+    const emailHtml = await renderPromise(res, 'templates/otp-email', {
       appName: res.app.locals.title,
       appUrl: config.app.appUrl,
       otp: otp,
     })
-  } catch (renderErr) {
-    logger.error(getRequestIp(req), req.url, req.headers, renderErr)
-    return res
-      .status(HttpStatus.INTERNAL_SERVER_ERROR)
-      .send(
-        'Error rendering OTP. Please try again later and if the problem persists, contact us.',
-      )
-  }
-  let mailOptions = {
-    to: email,
-    from: config.mail.mailer.from,
-    subject: 'One-Time Password (OTP) for ' + res.app.locals.title,
-    html: emailHTML,
-    headers: {
-      [EMAIL_HEADERS.emailType]: EMAIL_TYPES.loginOtp,
-    },
-  }
-  try {
-    await sendNodeMail({
-      mail: mailOptions,
-      options: { mailId: 'OTP' },
-    })
-    logger.info(`Login OTP sent:\temail=${email} ip=${getRequestIp(req)}`)
-    return res.status(HttpStatus.OK).send('OTP sent to ' + email + '!')
-  } catch (emailErr) {
-    logger.error(
-      'Mail otp error',
-      getRequestIp(req),
-      req.url,
-      req.headers,
-      emailErr,
-    )
+
+    await MailService.sendLoginOtp(recipient, emailHtml)
```

Comment:
could we move the `emailHtml` logic into `sendLoginOtp`, and thereby refactor the function signature to `sendLoginOtp(recipient, otp)`

Reply:
That's what I was discussing with you about. The current way is less than ideal because we use `req.app` to generate a HTML template instead of generating it without `req`; and we don't want to pass `req` into services.

I'll try out hardcoding the html in `templates` instead of using `ejs` engine.

Reply:
Only loginOtp and verificationOtp HTMLs can be hardcoded, as the rest requires some form of dynamicity which is most easily done with ejs. :(

Done for `sendLoginOtp`.